backend/inspirehep/records/receivers.py

```diff
@@ -41,3 +42,4 @@ def index_after_commit(sender, changes):
                     and current_app.config["FEATURE_FLAG_ENABLE_AUTHOR_DISAMBIGUATION"]
                 ):
                     disambiguate_authors.delay(str(model_instance.id))
+                redirect_references_to_merged_record.delay(str(model_instance.id))
```

Comment:
so one extra very rarely needed task per record update. Is that really how we want this to be done? (cc @drjova @pazembrz). Maybe we should have a single task instead that does both indexing and this?

Reply:
IMHO as this will be a rare task so optimization is not really important here, so it's better to keep it separate. We don't want bloated celery tasks, and it will be easier to keep logic of indexing and post-redirection data handling separate.

Reply:
It's not a rare task, it will be scheduled on every record update, but the task itself will do nothing most of the time.

Reply:
Maybe let's run the task only if the `deleted_records` from `model_instance` is not empty? I don't know how often it happens that records have `deleted_records`, but it will reduce somehow the number of calls of this task

Reply:
that's a great idea :+1: We have ~10k Literature records that have data in `deleted_records`: https://inspirehep.net/literature?sort=mostrecent&size=25&page=1&q=deleted_records%3A%2A, so it will be a big difference.